main/lsp/requests/hover.cc

```diff
@@ -24,11 +24,38 @@ string methodSignatureString(const core::GlobalState &gs, const core::TypePtr &r
     return contents;
 }
 
-unique_ptr<MarkupContent> formatRubyCode(MarkupKind markupKind, string str) {
-    if (markupKind == MarkupKind::Markdown && str.length() > 0) {
-        str = fmt::format("```ruby\n{}\n```", str);
+unique_ptr<MarkupContent> formatRubyCode(MarkupKind markupKind, string_view sigString,
+                                         optional<string_view> optionalDocString) {
+    string_view docString = optionalDocString.value_or("");
+    // Get rid of at most 1 trailing newline
```

Comment:
Any type of `docString` trimming should be in `findDocumentation`, not `formatRubyCode`.

Reply:
Also, you trim ASCII whitespace from each line in `findDocumentation`:

```
string_view line = absl::StripAsciiWhitespace(*it);
```

Does this conditional ever happen?

Reply:
Makes sense. I think I'll ensure exactly one newline.

Now that I think about it, I don't think I should be passing in documentation into `formatRubyCode`. I'm thinking of making a separate function to concatenate sigs and docs and return MarkupContent. wdyt?

Reply:
For now I trimmed the documentation in newline. This actually saved quite a few lines of code in `formatRubyCode`, thank you!

2df9e0e63